bazel/defs.bzl

```diff
@@ -93,6 +96,12 @@ def proxy_wasm_cc_binary(
             "--js-library=$(location @proxy_wasm_cpp_sdk//:proxy_wasm_intrinsics_js)",
             "-sSTANDALONE_WASM",
             "-sEXPORTED_FUNCTIONS=_malloc",
+            # Disable pthreads, because USE_PTHREADS implies SHARED_MEMORY and
+            #  "STANDALONE_WASM does not support shared memories yet"
+            #
+            # https://emscripten.org/docs/porting/pthreads.html
+            # https://emscripten.org/docs/api_reference/wasm_workers.html
+            "-sUSE_PTHREADS=0",
```

Comment:
Why is this needed? This setting is disabled by default in Emscripten v3.1.7 ([link](https://github.com/emscripten-core/emscripten/blob/48a16209b1a0de5efd8112ce6430415730008d18/src/settings.js#L1475)) and deprecated recently.

Reply:
Interesting! I'll dig into why I needed to provide this.

Reply:
Probably internal version is a bit different? Also, I believe that the more recent versions of Emscripten enable threads based on the `-pthread` flag, but then `USE_PTHREADS` is deprecated so it might not work around that in the future... if that's the case, then you might need to filter it out from arguments.

Also, updating Emscripten here would be a good idea.

Reply:
Ah, we need to change Platform (CPU, OS) to avoid Abseil and RE2 adding -pthread:
- https://github.com/search?q=repo%3Aabseil%2Fabseil-cpp%20%22-pthread%22&type=code
- https://github.com/search?q=repo%3Agoogle%2Fre2+%22-pthread%22&type=code

Working on that...

Reply:
Hmmm specifying `--platforms` (in command line or transition) didn't help.

Debugging ([docs](https://bazel.build/docs/configurable-attributes#faq-select-choose-condition)) with `bazelisk config` tells me we are targeting `@emsdk//:platform_wasm`. That may need to add target `@platforms//os:wasi`, and I also need to understand what's going on with `attr.threads` in [wasm_cc_binary.bzl](https://github.com/emscripten-core/emsdk/blob/main/bazel/emscripten_toolchain/wasm_cc_binary.bzl).

```
FragmentOptions com.google.devtools.build.lib.analysis.PlatformOptions {
  ...
  platform_mappings: null
  platforms: [@emsdk//:platform_wasm]
  target_platform_fallback: @bazel_tools//platforms:target_platform
```

Reply:
> specifying --platforms (in command line or transition) didn't help

And that's because Emscripten uses its own transition to implement wasm_cc_binary: https://github.com/emscripten-core/emsdk/blob/a896e3d066448b3530dbcaa48869fafefd738f57/bazel/emscripten_toolchain/wasm_cc_binary.bzl#L31-L40

Reply:
OK, so the issue is between [RE2 select](https://github.com/google/re2/blob/e66463312e1d30d427bbde6c40e7fd627dcfb82e/BUILD.bazel#L57-L70) and [emsdk platform](https://github.com/emscripten-core/emsdk/blob/a896e3d066448b3530dbcaa48869fafefd738f57/bazel/BUILD#L81-L86).

We can:
1. add `@platforms//os:wasi` to the emsdk platform
2. allow a platform override in wasm_cc_binary
3. change RE2 linkopts to select on wasm CPU or some wasm standalone trigger

Reply:
I merged a fix into emsdk. Now struggling with fallout from the emsdk upgrade.